### ISSUE TITLE
raise Exception from self leads to reported error

### ISSUE THREAD

**Issue by audoh-tickitto** [microsoft/pylance-release#806]
## Environment data

-   Language Server version: 2020.12.2
-   OS and version: Windows 10 Pro, 20H2, 19042.685
-   Python version (& distribution if applicable, e.g. Anaconda): 3.9.0 acquired via pyenv

## Expected behaviour

No error should be reported.

## Actual behaviour

The following error is reported:
```
Expected exception object or None
  "FooError" does not derive from BaseException
```

## Logs

```
N/A
```

## Code Snippet / Additional information

```python
class FooError(BaseException):
    def bar(self) -> None:
        raise Exception("baz") from self
```

The same also happens if extending any subclass of BaseException.

**Comment by judej** [comment]
@erictraut, would you please take a look a this?
thanks

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.1.0, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202110-6-january-2021
